src/lib/services/move/resourceService.ts

```diff
@@ -3,18 +3,82 @@ import { useQuery } from "@tanstack/react-query";
 
 import { CELATONE_QUERY_KEYS, useBaseApiRoute } from "lib/app-provider";
 import type { MoveAccountAddr } from "lib/types";
-import type { InternalResource } from "lib/types/move/resource";
+import type {
+  ResourceGroup,
+  ResourceGroupByAccount,
+} from "lib/types/move/resource";
+import { truncate } from "lib/utils";
 
 import { getAccountResources } from "./resource";
 
+export interface AccountResourcesReturn {
+  groupedByOwner: ResourceGroupByAccount[];
+  groupedByName: ResourceGroup[];
+  totalCount: number;
+}
+
 export const useAccountResources = ({
   address,
 }: {
   address: MoveAccountAddr;
-}): UseQueryResult<InternalResource[]> => {
+}): UseQueryResult<AccountResourcesReturn> => {
   const baseEndpoint = useBaseApiRoute("rest");
-  const queryFn: QueryFunction<InternalResource[]> = () =>
-    getAccountResources(baseEndpoint, address);
+  const queryFn: QueryFunction<AccountResourcesReturn> = () =>
+    getAccountResources(baseEndpoint, address).then((resources) => {
+      const groupedByOwner = resources.reduce<
+        Record<string, ResourceGroupByAccount>
+      >((acc, resource) => {
+        const [ownerName, groupName] = resource.structTag.split("::");
```

Comment:
I think we already have utility function for spliting path.

Reply:
existing function is for module and cannot apply to this one because groupName cannot be undefined, i will leave it like this for now